scripts/validate-requirements.sh

```diff
@@ -0,0 +1,274 @@
+GIT_VERSION="2.25.1"
+
+# Input variables
+END_USER_CREDENTIAL=""
+ORGANIZATION_ID=""
+BILLING_ACCOUNT=""
+
+# Collect the errors
+ERRORS=""
+# -------------------------- Funcions ---------------------------
+
+# Compare two semantic versions
+function compare_version(){
+    if [[ "$1" == "$2" ]]; then
+        echo 0
+        return 0
+    fi
+
+    local IFS=.
+    local i
+    local version1=("$1")
+    local version2=("$2")
+    for ((i=${#version1[@]}; i<${#version2[@]}; i++))
+    do
+        version1[i]=0
+    done
+    for ((i=0; i<${#version1[@]}; i++))
+    do
+        if [[ -z ${version2[i]} ]]
+        then
+            version2[i]=0
+        fi
+        if [[ ${version1[i]} > ${version2[i]} ]]
+        then
+            echo 1
+            return 1
+        fi
+        if [[ ${version1[i]} < ${version2[i]} ]]
+        then
+            echo 2
+            return 2
+        fi
+    done
+    return 0
+}
+
+# Validate the Terraform installation and version
+function validate_terraform(){
+
+    if [ ! "$(command -v terraform )" ]; then
+        echo "Terraform not found."
+        echo "Visit https://learn.hashicorp.com/tutorials/terraform/install-cli and follow the instructions to install Terraform."
+        ERRORS+=$'Terraform not found\n'
+    else
+        TERRAFORM_CURRENT_VERSION=$(terraform version -json | jq -r .terraform_version)
+        if [ "$(compare_version "$TERRAFORM_CURRENT_VERSION" "$TF_VERSION")" -ne 0 ]; then
+            echo "An incompatibly was found in Terraform version."
+            echo "Terraform version $TF_VERSION is required."
+            echo "Visit https://learn.hashicorp.com/tutorials/terraform/install-cli and follow the instructions to install Terraform."
+            ERRORS+=$'Terraform version is incompatible.\n'
+        fi
+    fi
+}
+
+# Validate the Google Cloud SDK installation and version
+function validate_gcloud(){
+    if [ ! "$(command -v gcloud)" ]; then
+        echo "gcloud CLI not found."
+        echo "Visit https://cloud.google.com/sdk/docs/install and follow the instructions to install gcloud CLI."
+        ERRORS+=$'gcloud not found.\n'
+    else
+        GCLOUD_CURRENT_VERSION=$(gcloud version --format=json | jq -r '."Google Cloud SDK"')
+        if [ "$(compare_version "$GCLOUD_CURRENT_VERSION" "$GCLOUD_SDK_VERSION")" -eq 2 ]; then
+            echo "An incompatibly was found in gcloud version."
+            echo "Version required is at least $GCLOUD_SDK_VERSION"
+            echo "Visit https://cloud.google.com/sdk/docs/install and follow the instructions to install gcloud CLI."
+            ERRORS+=$'gcloud version is incompatible.\n'
+        fi
+    fi
+}
+
+# Validate the Git installation and version
+function validate_git(){
+    if [ ! "$(command -v git)" ]; then
+        echo "git not found."
+        echo "Visit https://git-scm.com/book/en/v2/Getting-Started-Installing-Git and follow the instructions to install Git."
+        ERRORS+=$'git not found.\n'
```

Comment:
Can we exit early in these cases? We prefer early exit as opposed to wrapping in else making it less readable.

Reply:
Here and thoughout.

Reply:
I've tried to follow the same idea we have for some CFT scripts, for example the one that realizes the lint tests (make docker_test_lint) so the user knows exactly what is missing on a single run. 
I've just added an exit for JQ for example and now when gcloud is missing, it doesn't run the validation for gcloud local config neither roles check (would fail anyway), so it makes the errors output way shorter, basically will only aggregate when is about Terraform, git and gcloud installation issues.

Reply:
sgtm